Raspberry Pi OS build failure in CI - `Error: No such container:path:`

The current 1.0.1 tag failed to build a Raspberry Pi OS image in CI with the [error](https://github.com/WebThingsIO/gateway/actions/runs/3489095092/jobs/5838772888#step:5:13059):

`Error: No such container:path: pigen_work:/pi-gen/work/2022-11-17-Raspbian/stage3/rootfs/home/pi/webthings/gateway`

Longer output:

```
2022-11-17T16:11:52.1487537Z Running export stage for 2022-11-17-Raspbian
2022-11-17T16:11:52.1611643Z Creating new base 2022-11-17-Raspbian.qcow2 from image-stage0.qcow2
2022-11-17T16:11:58.9350703Z Rebasing image-stage1.qcow2 onto 2022-11-17-Raspbian.qcow2
2022-11-17T16:11:58.9429071Z Rebasing image-stage2.qcow2 onto image-stage1.qcow2
2022-11-17T16:11:58.9492819Z Rebasing image-stage3.qcow2 onto image-stage2.qcow2
2022-11-17T16:11:58.9574245Z Committing image-stage3.qcow2 to 2022-11-17-Raspbian.qcow2
2022-11-17T16:11:58.9687725Z     (0.00/100%)
2022-11-17T16:11:58.9696699Z     (0.00/100%)
2022-11-17T16:11:58.9704278Z     (0.00/100%)
2022-11-17T16:11:58.9707141Z     (0.00/100%)
2022-11-17T16:11:58.9716531Z     (0.00/100%)
2022-11-17T16:11:58.9729100Z     (0.00/100%)
2022-11-17T16:11:58.9732808Z     (0.00/100%)
2022-11-17T16:11:58.9738791Z     (0.00/100%)
2022-11-17T16:11:58.9749702Z     (0.00/100%)
2022-11-17T16:11:58.9750471Z     (0.00/100%)
2022-11-17T16:11:58.9759948Z     (0.00/100%)
2022-11-17T16:11:58.9765807Z     (0.00/100%)
2022-11-17T16:11:58.9770334Z     (0.00/100%)
2022-11-17T16:11:58.9776192Z     (0.00/100%)
2022-11-17T16:11:58.9788001Z     (0.00/100%)
2022-11-17T16:11:58.9790752Z     (0.00/100%)
2022-11-17T16:11:58.9800081Z     (0.00/100%)
2022-11-17T16:11:58.9807445Z     (0.00/100%)
2022-11-17T16:11:58.9809935Z     (0.00/100%)
2022-11-17T16:11:58.9816086Z     (0.00/100%)
2022-11-17T16:11:58.9825224Z     (0.00/100%)
2022-11-17T16:11:58.9828409Z     (0.00/100%)
2022-11-17T16:11:58.9835559Z     (0.00/100%)
2022-11-17T16:11:58.9846653Z     (0.00/100%)
2022-11-17T16:11:58.9847068Z     (0.00/100%)
2022-11-17T16:11:58.9854014Z     (0.00/100%)
2022-11-17T16:11:58.9869998Z     (0.00/100%)
2022-11-17T16:11:58.9872348Z     (0.00/100%)
2022-11-17T16:11:58.9881637Z     (0.00/100%)
2022-11-17T16:11:58.9890073Z     (0.00/100%)
2022-11-17T16:11:58.9890413Z     (0.00/100%)
2022-11-17T16:11:58.9900685Z     (0.00/100%)
2022-11-17T16:11:58.9908282Z     (0.00/100%)
2022-11-17T16:11:58.9910032Z     (0.00/100%)
2022-11-17T16:11:58.9920816Z     (0.00/100%)
2022-11-17T16:11:58.9928573Z     (0.00/100%)
2022-11-17T16:11:58.9929922Z     (0.00/100%)
2022-11-17T16:11:58.9941322Z     (0.00/100%)
2022-11-17T16:11:58.9949813Z     (0.00/100%)
2022-11-17T16:11:58.9950248Z     (0.00/100%)
2022-11-17T16:11:58.9965275Z     (0.00/100%)
2022-11-17T16:11:58.9974238Z     (0.00/100%)
2022-11-17T16:11:58.9974603Z     (0.00/100%)
2022-11-17T16:11:58.9996767Z     (0.00/100%)
2022-11-17T16:11:59.0050000Z     (0.00/100%)
2022-11-17T16:11:59.0050547Z     (0.00/100%)
2022-11-17T16:11:59.0050779Z     (0.00/100%)
2022-11-17T16:11:59.0094602Z     (0.00/100%)
2022-11-17T16:11:59.0096198Z     (0.00/100%)
2022-11-17T16:11:59.0096465Z     (0.00/100%)
2022-11-17T16:11:59.0096673Z     (0.00/100%)
2022-11-17T16:11:59.0096885Z     (0.00/100%)
2022-11-17T16:11:59.0097285Z     (0.00/100%)
2022-11-17T16:11:59.0097505Z     (0.00/100%)
2022-11-17T16:11:59.0097817Z     (0.00/100%)
2022-11-17T16:11:59.0098055Z     (0.00/100%)
2022-11-17T16:11:59.0098274Z     (0.00/100%)
2022-11-17T16:11:59.0098471Z     (0.00/100%)
2022-11-17T16:11:59.0101297Z     (0.00/100%)
2022-11-17T16:11:59.0114621Z     (0.00/100%)
2022-11-17T16:11:59.0115173Z     (0.00/100%)
2022-11-17T16:11:59.0125388Z     (0.00/100%)
2022-11-17T16:11:59.0134940Z     (0.00/100%)
2022-11-17T16:11:59.0135401Z     (0.00/100%)
2022-11-17T16:11:59.0145030Z     (0.00/100%)
2022-11-17T16:11:59.0153105Z     (0.00/100%)
2022-11-17T16:11:59.0157788Z     (0.00/100%)
2022-11-17T16:11:59.0168552Z     (0.00/100%)
2022-11-17T16:11:59.0181267Z     (0.00/100%)
2022-11-17T16:11:59.0183580Z     (0.00/100%)
2022-11-17T16:11:59.0197546Z     (0.00/100%)
2022-11-17T16:11:59.0208729Z     (0.00/100%)
2022-11-17T16:11:59.0209175Z     (0.00/100%)
2022-11-17T16:11:59.0217519Z     (0.00/100%)
2022-11-17T16:11:59.0217937Z     (0.00/100%)
2022-11-17T16:11:59.0219053Z     (0.00/100%)
2022-11-17T16:11:59.0219277Z     (0.00/100%)
2022-11-17T16:11:59.0219654Z     (0.00/100%)
2022-11-17T16:11:59.1209164Z     (0.00/100%)
2022-11-17T16:11:59.4151133Z     (1.14/100%)
2022-11-17T16:11:59.4820034Z     (2.15/100%)
2022-11-17T16:11:59.5583446Z     (3.16/100%)
2022-11-17T16:11:59.6679755Z     (4.20/100%)
2022-11-17T16:11:59.7216027Z     (5.21/100%)
2022-11-17T16:11:59.8231950Z     (6.30/100%)
2022-11-17T16:12:00.1303618Z     (7.43/100%)
2022-11-17T16:12:00.4894046Z     (8.54/100%)
2022-11-17T16:12:00.5691215Z     (9.67/100%)
2022-11-17T16:12:00.6415395Z     (10.76/100%)
2022-11-17T16:12:00.7277891Z     (11.89/100%)
2022-11-17T16:12:00.7843775Z     (13.02/100%)
2022-11-17T16:12:01.0359269Z     (14.15/100%)
2022-11-17T16:12:01.8869292Z     (15.27/100%)
2022-11-17T16:12:01.9366183Z     (16.27/100%)
2022-11-17T16:12:01.9775269Z     (17.40/100%)
2022-11-17T16:12:02.0467752Z     (18.53/100%)
2022-11-17T16:12:04.1838438Z     (19.54/100%)
2022-11-17T16:12:04.2097721Z     (20.66/100%)
2022-11-17T16:12:04.2580960Z     (21.74/100%)
2022-11-17T16:12:04.3309615Z     (22.82/100%)
2022-11-17T16:12:04.3444339Z     (23.87/100%)
2022-11-17T16:12:04.3781494Z     (24.99/100%)
2022-11-17T16:12:05.0348134Z     (26.02/100%)
2022-11-17T16:12:05.0453372Z     (27.07/100%)
2022-11-17T16:12:05.0865687Z     (28.15/100%)
2022-11-17T16:12:05.3410098Z     (29.27/100%)
2022-11-17T16:12:05.3720824Z     (30.34/100%)
2022-11-17T16:12:05.4282425Z     (31.36/100%)
2022-11-17T16:12:05.4885314Z     (32.45/100%)
2022-11-17T16:12:05.5313698Z     (33.54/100%)
2022-11-17T16:12:05.5537035Z     (34.56/100%)
2022-11-17T16:12:06.5289073Z     (35.57/100%)
2022-11-17T16:12:06.5831472Z     (36.58/100%)
2022-11-17T16:12:06.6201500Z     (37.66/100%)
2022-11-17T16:12:06.8215732Z     (38.68/100%)
2022-11-17T16:12:06.9196917Z     (39.76/100%)
2022-11-17T16:12:06.9343380Z     (40.87/100%)
2022-11-17T16:12:07.1199087Z     (41.91/100%)
2022-11-17T16:12:07.1820986Z     (42.94/100%)
2022-11-17T16:12:07.2662902Z     (43.96/100%)
2022-11-17T16:12:07.2904932Z     (45.09/100%)
2022-11-17T16:12:07.4867438Z     (46.18/100%)
2022-11-17T16:12:07.5369634Z     (47.23/100%)
2022-11-17T16:12:07.5792761Z     (48.34/100%)
2022-11-17T16:12:07.6906317Z     (49.39/100%)
2022-11-17T16:12:07.7243235Z     (50.49/100%)
2022-11-17T16:12:07.7770706Z     (51.49/100%)
2022-11-17T16:12:07.8245839Z     (52.62/100%)
2022-11-17T16:12:07.8445470Z     (53.69/100%)
2022-11-17T16:12:09.9366804Z     (54.80/100%)
2022-11-17T16:12:10.0260148Z     (55.91/100%)
2022-11-17T16:12:10.2654255Z     (56.92/100%)
2022-11-17T16:12:10.3735021Z     (57.92/100%)
2022-11-17T16:12:10.4346695Z     (58.96/100%)
2022-11-17T16:12:10.4825388Z     (60.01/100%)
2022-11-17T16:12:10.5311955Z     (61.08/100%)
2022-11-17T16:12:10.5757647Z     (62.14/100%)
2022-11-17T16:12:10.8326500Z     (63.23/100%)
2022-11-17T16:12:10.9308354Z     (64.27/100%)
2022-11-17T16:12:10.9831801Z     (65.37/100%)
2022-11-17T16:12:11.0770262Z     (66.38/100%)
2022-11-17T16:12:11.1915459Z     (67.41/100%)
2022-11-17T16:12:11.2755131Z     (68.44/100%)
2022-11-17T16:12:11.3319543Z     (69.55/100%)
2022-11-17T16:12:11.5286322Z     (70.69/100%)
2022-11-17T16:12:11.5938361Z     (71.72/100%)
2022-11-17T16:12:11.6733176Z     (72.73/100%)
2022-11-17T16:12:13.4340752Z     (73.84/100%)
2022-11-17T16:12:13.6267054Z     (74.92/100%)
2022-11-17T16:12:13.7772656Z     (75.92/100%)
2022-11-17T16:12:13.8675989Z     (77.04/100%)
2022-11-17T16:12:13.9844148Z     (78.07/100%)
2022-11-17T16:12:14.0975096Z     (79.23/100%)
2022-11-17T16:12:14.1758237Z     (80.26/100%)
2022-11-17T16:12:15.0803524Z     (81.37/100%)
2022-11-17T16:12:15.1862572Z     (82.49/100%)
2022-11-17T16:12:15.2820864Z     (83.50/100%)
2022-11-17T16:12:15.3191769Z     (84.54/100%)
2022-11-17T16:12:15.4246728Z     (85.54/100%)
2022-11-17T16:12:15.4822119Z     (86.56/100%)
2022-11-17T16:12:15.7237302Z     (87.61/100%)
2022-11-17T16:12:17.1782463Z     (88.63/100%)
2022-11-17T16:12:17.2649188Z     (89.69/100%)
2022-11-17T16:12:17.3780169Z     (90.83/100%)
2022-11-17T16:12:17.4773796Z     (91.89/100%)
2022-11-17T16:12:17.5725318Z     (92.95/100%)
2022-11-17T16:12:17.6220992Z     (94.02/100%)
2022-11-17T16:12:17.7355356Z     (95.04/100%)
2022-11-17T16:12:18.3462777Z     (96.16/100%)
2022-11-17T16:12:18.5781132Z     (97.21/100%)
2022-11-17T16:12:18.9401043Z     (98.24/100%)
2022-11-17T16:12:18.9706760Z     (99.33/100%)
2022-11-17T16:12:18.9743656Z     (100.00/100%)
2022-11-17T16:12:19.0675247Z     (100.00/100%)
2022-11-17T16:12:19.0675550Z     (100.00/100%)
2022-11-17T16:12:19.0675801Z     (100.00/100%)
2022-11-17T16:12:19.0676034Z     (100.00/100%)
2022-11-17T16:12:19.0704395Z     (100.00/100%)
2022-11-17T16:12:19.0711977Z     (100.00/100%)
2022-11-17T16:12:19.0717117Z     (100.00/100%)
2022-11-17T16:12:19.0719228Z Image committed.
2022-11-17T16:12:19.0736758Z Rebasing back image-stage1.qcow2 onto image-stage0.qcow2
2022-11-17T16:12:19.0798954Z Rebasing back image-stage2.qcow2 onto image-stage1.qcow2
2022-11-17T16:12:19.0858026Z Rebasing back image-stage3.qcow2 onto image-stage2.qcow2
2022-11-17T16:12:19.0954556Z Mounting image /pi-gen/work/Raspbian/export-image/2022-11-17-Raspbian.qcow2 to rootfs /pi-gen/work/Raspbian/export-image/rootfs
2022-11-17T16:12:19.1513670Z mount: /dev/mapper/nbd0p2 mounted on /pi-gen/work/Raspbian/export-image/rootfs.
2022-11-17T16:12:19.1644458Z mount: /dev/mapper/nbd0p1 mounted on /pi-gen/work/Raspbian/export-image/rootfs/boot.
2022-11-17T16:12:19.1666489Z [16:12:19] Begin /pi-gen/export-image
2022-11-17T16:12:19.1702550Z [16:12:19] Begin /pi-gen/export-image/prerun.sh
2022-11-17T16:12:19.1775701Z [16:12:19] End /pi-gen/export-image/prerun.sh
2022-11-17T16:12:19.1797401Z [16:12:19] Begin /pi-gen/export-image/00-allow-rerun
2022-11-17T16:12:19.1844864Z [16:12:19] Begin /pi-gen/export-image/00-allow-rerun/00-run.sh
2022-11-17T16:12:19.2305163Z [16:12:19] End /pi-gen/export-image/00-allow-rerun/00-run.sh
2022-11-17T16:12:19.2391815Z [16:12:19] End /pi-gen/export-image/00-allow-rerun
2022-11-17T16:12:19.2487031Z [16:12:19] Begin /pi-gen/export-image/01-set-sources
2022-11-17T16:12:19.2512387Z [16:12:19] Begin /pi-gen/export-image/01-set-sources/01-run.sh
2022-11-17T16:12:20.7106625Z Get:1 http://archive.raspberrypi.org/debian buster InRelease [32.6 kB]
2022-11-17T16:12:20.7929952Z Get:2 http://raspbian.raspberrypi.org/raspbian buster InRelease [15.0 kB]
2022-11-17T16:12:23.2571151Z Get:3 http://raspbian.raspberrypi.org/raspbian buster/main armhf Packages [13.0 MB]
2022-11-17T16:12:24.9763729Z Get:4 http://raspbian.raspberrypi.org/raspbian buster/contrib armhf Packages [58.8 kB]
2022-11-17T16:12:25.2231974Z Get:5 http://archive.raspberrypi.org/debian buster/main armhf Packages [392 kB]
2022-11-17T16:12:25.4130245Z Get:6 http://raspbian.raspberrypi.org/raspbian buster/non-free armhf Packages [104 kB]
2022-11-17T16:12:25.9588141Z Get:7 http://raspbian.raspberrypi.org/raspbian buster/rpi armhf Packages [1360 B]
2022-11-17T16:12:31.9246706Z Fetched 13.6 MB in 12s (1141 kB/s)
2022-11-17T16:12:38.9952657Z Reading package lists...
2022-11-17T16:12:39.4340391Z Reading package lists...
2022-11-17T16:12:40.8062900Z Building dependency tree...
2022-11-17T16:12:40.8158900Z Reading state information...
2022-11-17T16:12:41.2521455Z Calculating upgrade...
2022-11-17T16:12:41.8369466Z 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
2022-11-17T16:12:41.9768599Z [16:12:41] End /pi-gen/export-image/01-set-sources/01-run.sh
2022-11-17T16:12:41.9836904Z [16:12:41] End /pi-gen/export-image/01-set-sources
2022-11-17T16:12:41.9852600Z [16:12:41] Begin /pi-gen/export-image/02-network
2022-11-17T16:12:41.9885490Z [16:12:41] Begin /pi-gen/export-image/02-network/01-run.sh
2022-11-17T16:12:41.9962064Z [16:12:41] End /pi-gen/export-image/02-network/01-run.sh
2022-11-17T16:12:42.0030050Z [16:12:42] End /pi-gen/export-image/02-network
2022-11-17T16:12:42.0047187Z [16:12:42] Begin /pi-gen/export-image/03-set-partuuid
2022-11-17T16:12:42.0065583Z [16:12:42] Begin /pi-gen/export-image/03-set-partuuid/00-run.sh
2022-11-17T16:12:42.0119842Z [16:12:42] End /pi-gen/export-image/03-set-partuuid/00-run.sh
2022-11-17T16:12:42.0187785Z [16:12:42] End /pi-gen/export-image/03-set-partuuid
2022-11-17T16:12:42.0206604Z [16:12:42] Begin /pi-gen/export-image/04-finalise
2022-11-17T16:12:42.0226258Z [16:12:42] Begin /pi-gen/export-image/04-finalise/01-run.sh
2022-11-17T16:12:42.2188759Z Stopping fake hwclock: saving system time.
2022-11-17T16:12:43.5147007Z Mode:     real
2022-11-17T16:12:43.5147284Z Files:    3824
2022-11-17T16:12:43.5147521Z Linked:   791 files
2022-11-17T16:12:43.5147771Z Compared: 0 xattrs
2022-11-17T16:12:43.5148002Z Compared: 1497 files
2022-11-17T16:12:43.5148249Z Saved:    51.03 MiB
2022-11-17T16:12:43.5148487Z Duration: 1.16 seconds
2022-11-17T16:12:44.6213509Z /pi-gen/work/Raspbian/export-image/rootfs: 9.2 GiB (9853550592 bytes) trimmed
2022-11-17T16:12:44.8172412Z /pi-gen/work/Raspbian/export-image/rootfs/sys
2022-11-17T16:12:44.8212613Z /pi-gen/work/Raspbian/export-image/rootfs/proc
2022-11-17T16:12:44.8255722Z /pi-gen/work/Raspbian/export-image/rootfs/dev/pts
2022-11-17T16:12:44.8296757Z /pi-gen/work/Raspbian/export-image/rootfs/dev
2022-11-17T16:12:44.8336564Z /pi-gen/work/Raspbian/export-image/rootfs/boot
2022-11-17T16:12:44.8644964Z /pi-gen/work/Raspbian/export-image/rootfs
2022-11-17T16:12:44.9358679Z /dev/nbd0 disconnected
2022-11-17T16:12:44.9359028Z Connect block device to source qcow2
2022-11-17T16:12:44.9914848Z Resize fs and partition
2022-11-17T16:12:45.0388207Z e2fsck 1.44.5 (15-Dec-2018)
2022-11-17T16:12:45.0780570Z Pass 1: Checking inodes, blocks, and sizes
2022-11-17T16:12:46.1025723Z Pass 2: Checking directory structure
2022-11-17T16:12:46.5233113Z Pass 3: Checking directory connectivity
2022-11-17T16:12:46.5256974Z Pass 4: Checking reference counts
2022-11-17T16:12:46.5580440Z Pass 5: Checking group summary information
2022-11-17T16:12:46.6313512Z rootfs: 108750/771552 files (0.1% non-contiguous), 658170/3080192 blocks
2022-11-17T16:12:46.6612882Z resize2fs 1.44.5 (15-Dec-2018)
2022-11-17T16:12:46.6660347Z dumpe2fs 1.44.5 (15-Dec-2018)
2022-11-17T16:12:46.6706273Z resize2fs 1.44.5 (15-Dec-2018)
2022-11-17T16:12:47.0864818Z Resizing the filesystem on /dev/mapper/nbd0p2 to 694277 (4k) blocks.
2022-11-17T16:12:47.0865314Z Begin pass 2 (max = 490224)
2022-11-17T16:13:17.7002279Z Relocating blocks             ----------------------------------------XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-11-17T16:13:17.7002826Z Begin pass 3 (max = 94)
2022-11-17T16:13:19.1659696Z Scanning inode table          ----------------------------------------XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-11-17T16:13:19.1660225Z Begin pass 4 (max = 16080)
2022-11-17T16:13:29.1150717Z Updating inode references     ----------------------------------------XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
2022-11-17T16:13:29.1151380Z The filesystem on /dev/mapper/nbd0p2 is now 694277 (4k) blocks long.
2022-11-17T16:13:29.1151611Z 
2022-11-17T16:13:30.2534830Z Disconnect block device
2022-11-17T16:13:30.2921938Z /dev/nbd0 disconnected
2022-11-17T16:13:30.2922627Z Shrinking qcow2 image
2022-11-17T16:13:30.8057996Z Image resized.
2022-11-17T16:13:30.8105276Z Convert qcow2 to raw image
2022-11-17T16:13:49.0781929Z Get PARTUUIDs from image
2022-11-17T16:13:49.0825254Z Boot: 31e49f6f-01
2022-11-17T16:13:49.0825599Z Root: 31e49f6f-02
2022-11-17T16:13:49.0825845Z Mount image
2022-11-17T16:13:49.1705107Z Setup PARTUUIDs
2022-11-17T16:13:49.1706002Z Set UUIDs to make it bootable
2022-11-17T16:13:49.1799716Z Umount image
2022-11-17T16:13:49.2469474Z Remove qcow2 export image
2022-11-17T16:16:14.1706889Z   adding: 2022-11-17-Raspbian.img (deflated 70%)
2022-11-17T16:16:14.1737942Z [16:16:14] End /pi-gen/export-image/04-finalise/01-run.sh
2022-11-17T16:16:14.1813807Z [16:16:14] End /pi-gen/export-image/04-finalise
2022-11-17T16:16:14.1834771Z [16:16:14] End /pi-gen/export-image
2022-11-17T16:16:14.3708805Z /pi-gen/work/Raspbian/export-image/rootfs: 21 GiB (22535958528 bytes) trimmed
2022-11-17T16:16:14.7204561Z /dev/nbd0 disconnected
2022-11-17T16:16:14.7206190Z [16:16:14] End /pi-gen
2022-11-17T16:16:14.7206642Z [16:16:14] Unloading image
2022-11-17T16:16:14.7207015Z sending incremental file list
2022-11-17T16:16:14.7207349Z build.log
2022-11-17T16:16:14.7207508Z 
2022-11-17T16:16:14.7207677Z sent 6,786 bytes  received 35 bytes  13,642.00 bytes/sec
2022-11-17T16:16:14.7208078Z total size is 6,691  speedup is 0.98
2022-11-17T16:16:15.0283184Z 
2022-11-17T16:16:15.0290754Z real	112m33.822s
2022-11-17T16:16:15.0291144Z copying results from deploy/
2022-11-17T16:16:15.0292750Z user	0m0.762s
2022-11-17T16:16:15.0293028Z sys	0m0.432s
2022-11-17T16:16:18.9757491Z total 902M
2022-11-17T16:16:18.9758627Z drwxr-xr-x  2 runner docker 4.0K Nov 17 16:16 .
2022-11-17T16:16:18.9759616Z drwxr-xr-x 12 runner docker 4.0K Nov 17 16:16 ..
2022-11-17T16:16:18.9760274Z -rw-r--r--  1 runner docker 100K Nov 17 16:12 2022-11-17-Raspbian.info
2022-11-17T16:16:18.9760877Z -rw-r--r--  1 runner docker 6.6K Nov 17 16:16 build.log
2022-11-17T16:16:18.9765561Z -rw-r--r--  1 runner docker 902M Nov 17 16:16 image_2022-11-17-Raspbian.zip
2022-11-17T16:16:18.9766406Z Done! Your image(s) should be in deploy/
2022-11-17T16:16:28.1308969Z Error: No such container:path: pigen_work:/pi-gen/work/2022-11-17-Raspbian/stage3/rootfs/home/pi/webthings/gateway
2022-11-17T16:16:28.1457978Z ##[error]Process completed with exit code 1.
2022-11-17T16:16:28.1604110Z Post job cleanup.
2022-11-17T16:16:28.4242822Z [command]/usr/bin/git version
2022-11-17T16:16:28.4388318Z git version 2.38.1
2022-11-17T16:16:28.4455788Z Temporarily overriding HOME='/home/runner/work/_temp/10658c9b-1c1a-46c6-82b8-a166c7935532' before making global git config changes
2022-11-17T16:16:28.4458767Z Adding repository directory to the temporary git global config as a safe directory
2022-11-17T16:16:28.4464394Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/gateway/gateway
2022-11-17T16:16:28.4533698Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2022-11-17T16:16:28.4579212Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
2022-11-17T16:16:28.5047774Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2022-11-17T16:16:28.5079940Z http.https://github.com/.extraheader
2022-11-17T16:16:28.5092410Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2022-11-17T16:16:28.5135848Z [command]/usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :
2022-11-17T16:16:28.5683314Z Cleaning up orphan processes
```

--- a/.github/workflows/prerelease.yml
+++ b/.github/workflows/prerelease.yml
@@ -52,7 +52,7 @@ jobs:
       - name: Install qemu
         run: |
           sudo apt update
-          sudo apt install -y qemu qemu-user-static binfmt-support --no-install-recommends
+          sudo apt install -y qemu qemu-user-static binfmt-support kpartx --no-install-recommends
       - name: Setup qemu-user-static
         run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
       - name: Build the image

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -51,7 +51,7 @@ jobs:
       - name: Install qemu
         run: |
           sudo apt update
-          sudo apt install -y qemu qemu-user-static binfmt-support --no-install-recommends
+          sudo apt install -y qemu qemu-user-static binfmt-support kpartx --no-install-recommends
       - name: Setup qemu-user-static
         run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
       - name: Build the image

--- a/image/build.sh
+++ b/image/build.sh
@@ -56,8 +56,15 @@ _image_name=${_image_name/.zip/}
 mv pi-gen/deploy/*.zip "gateway-${_version}.img.zip"
 shasum --algorithm 256 "gateway-${_version}.img.zip" > "gateway-${_version}.img.zip.sha256sum"
 
-# copy the built gateway out of the docker image
-docker cp "pigen_work:/pi-gen/work/${_image_name}/stage3/rootfs/home/pi/webthings/gateway" .
+# Unzip the gateway image to extract gateway directory for OTA update
+unzip gateway-${_version}.img.zip
+# Map partitions from image and read name of last partition which should be the Linux one
+_partition=$(sudo kpartx -av *-Raspbian.img | awk '{print $3}' | tail -n 1)
+# Mount the partition
+sudo mkdir -p /mnt/gateway-${_version}
+sudo mount /dev/mapper/${_partition} /mnt/gateway-${_version}
+# Copy the gateway folder out of the partition to the current directory
+cp -r /mnt/gateway-${_version}/home/pi/webthings/gateway .
 
 makeContentAddressedArchive() {
   base_name="$1"
@@ -73,5 +80,7 @@ makeContentAddressedArchive node_modules
 makeContentAddressedArchive gateway
 
 # clean up
-rm -rf gateway node_modules pi-gen
+sudo umount /mnt/gateway-${_version}
+sudo rm -rf /mnt/gateway-${_version}
+rm -rf gateway node_modules pi-gen *-Raspbian.img
 docker rm -v pigen_work

